app/src/org/commcare/network/CTInterceptorConfig.kt

```diff
@@ -0,0 +1,48 @@
+package org.commcare.network
+
+import com.appmattus.certificatetransparency.CTLogger
+import com.appmattus.certificatetransparency.VerificationResult
+import com.appmattus.certificatetransparency.certificateTransparencyInterceptor
+
+import okhttp3.OkHttpClient
+import org.commcare.core.network.HttpBuilderConfig
+import org.commcare.preferences.HiddenPreferences
+import org.commcare.util.LogTypes
+import org.javarosa.core.services.Logger
+
+/**
+ * Adds and removes Certificate Transparency (CT) network interceptor
+ *
+ */
+class CTInterceptorConfig:HttpBuilderConfig  {
+
+
+    override fun performCustomConfig(client: OkHttpClient.Builder): OkHttpClient.Builder {
+        if(HiddenPreferences.isCertificateTransparencyEnabled()) {
+            return client.addNetworkInterceptor(getInterceptor)
+        }
+
+        // In case there are CT Interceptors already attached
+        removeCTInterceptors(client)
```

Comment:
I don't think this is possible, since the builder starts fresh each time in `customizeRetrofitSetup()`, so I'd prefer to avoid the redundancy, and especially the class name reflection check. Did you test specifically to see if things were getting added multiple times without this?

Reply:
Yes, that's what I thought too but everything in the `CommCareNetworkServiceGenerator` is static and so we get the same `OkHttpClient.Builder` object with all the interceptors added previously. So this is just to remove the CT interceptors before the `build()` method is called. Unless I'm missing something.

Reply:
Ah, got it. 

That does seem tricky to deal with. I am a bit concerned about the structure of this approach being a straight string comparison, though. 

Could we move [this comparison](https://github.com/dimagi/commcare-android/pull/2681/files/0ac900e58e0b5d60b931e996deabf632b5d2cbde#diff-3ef7953a7a8b4016c33692a7f7330318e5e47b976fc5c2d05ad84efad0bd812bR39) to an `instanceof` check instead? That would prevent future drift issues that require a tight coupling of that string and the name

Reply:
We can't because the `CertificateTransparencyInterceptor` class is internal, but we can switch to an object comparison.